applications/luci-app-packet-capture/htdocs/luci-static/resources/view/packet_capture/tcpdump.js

```diff
@@ -0,0 +1,293 @@
+'use strict';
+'require rpc';
+'require uci';
+'require ui';
+'require fs';
+'require form';
+'require network';
+'require tools.widgets as widgets';
+
+var eventSource,
+	captureFilePoll,
+	hostName;
+
+function stopTcpdump() {
+	fs.exec("/usr/sbin/packet_capture_stop").then(function(replay) {
+		if(eventSource) { 
+			eventSource.close();
+		}
+	}.bind(this)).catch(function(error) {
+		console.log(error);
+	});
+}
+
+window.onbeforeunload = stopTcpdump;
+
+var callLuciProcessList = rpc.declare({
+	object: 'luci',
+	method: 'getProcessList',
+	expect: { result: [] }
+});
+
+var callInitAction = rpc.declare({
+	object: 'luci',
+	method: 'setInitAction',
+	params: [ 'name', 'action' ],
+	expect: { result: false }
+});
+
+function findButton(title) {
+	var buttons = document.querySelectorAll('button');
+	for (var i=0; i < buttons.length; i++) {
+		if (buttons[i].firstChild.nodeValue == title) {
+			return buttons[i];
+		}
+	}
+}
+
+function addOutput() {
+	var tcpdumpOut = document.querySelectorAll('[id$="tcpdump_out"]')[0];
+	if(tcpdumpOut) {
+		return;
+	}
+
+	var frameEl = E('div', {'class': 'cbi-value'});
+
+	frameEl.appendChild(E('textarea', {
+			'id': 'tcpdump_out',
+			'class': 'cbi-input-textarea',
+			'readonly': '',
+			'style': 'width:100%',
+			'rows': 30,
+	}));
+
+	frameEl.firstElementChild.style.fontFamily = 'monospace';
+
+	var downloadBtn = document.querySelectorAll('[id$="download_file"]')[0];
+	if(downloadBtn) {
+		downloadBtn.parentNode.insertBefore(frameEl, downloadBtn.nextSibling);
+	}
+}
+
+var downloadCaptureFile = function(ev) {
+	var form = E('form', {
+		method: 'post',
+		action: '/cgi-bin/cgi-download',
+		enctype: 'application/x-www-form-urlencoded'
+	}, E('input', { type: 'hidden', name: 'sessionid', value: rpc.getSessionID()},
+	   E('input', { type: 'hidden', name: 'path', value: "/tmp/capture.pcap"},
+	   E('input', { type: 'hidden', name: 'filename', value: hostName + "-" + Date.now() + ".pcap"}
+	))));
+
+	ev.currentTarget.parentNode.appendChild(form);
+	form.submit();
+	form.parentNode.removeChild(form);
+}
+
+function subscribeTcpdump() {
+	eventSource = new EventSource('/ubus/subscribe/tcpdump' + '?' + rpc.getSessionID());
+	eventSource.onerror = function(event) {
+		eventSource.close();
+		console.log(event);	
+	};
+
+	addOutput();
+	var textOut = document.querySelectorAll('[id$="tcpdump_out"]')[0];
+	textOut.value = "";
+	eventSource.addEventListener("tcpdump.data", function(event) {
+		textOut.value = textOut.value + "\n" + JSON.parse(event.data).data;
+	});
+}
+
+function updateButtons() {
+	var tasks = [];
+	tasks.push(callLuciProcessList().then(L.bind(function(list) {
+		for (var i = 0; i < list.length; i++) {
+			if(list[i].COMMAND.includes("/usr/sbin/packet_capture")) {
```

Comment:
For this and the subsequent `if` conditions:
 - add space after `if`
 - omit curly braces for single-statement conditionals